xbmc/dbwrappers/dataset.h

```diff
@@ -396,10 +396,30 @@ class Dataset  {
 /* Struct to store an indexMapped field access entry */
   struct FieldIndexMapEntry
   {
-    explicit FieldIndexMapEntry(const char* name) : fieldIndex(~0), strName(name) {}
-    bool operator<(const FieldIndexMapEntry& other) const { return strName < other.strName; }
-    unsigned int fieldIndex;
-    std::string strName;
+    explicit FieldIndexMapEntry(const char* name) : fieldIndex(~0), strName(strdup(name)) {}
+
+    FieldIndexMapEntry(const FieldIndexMapEntry& other)
+      : fieldIndex(other.fieldIndex), strName(strdup(other.strName))
+    {
+    }
+
+    FieldIndexMapEntry& operator=(const FieldIndexMapEntry& other)
+    {
+      fieldIndex = other.fieldIndex;
+      free(strName);
+      strName = strdup(other.strName);
+      return *this;
+    }
+
+    ~FieldIndexMapEntry() { free(strName); }
+
+    bool operator<(const FieldIndexMapEntry& other) const
+    {
+      return strncmp(strName, other.strName, strlen(other.strName)) < 0;
+    }
+
+    unsigned int fieldIndex{0};
```

Comment:
Looks like this should be initialized to `~0` and the initialization dropped from the constructor.

Reply:
I agree. Will  be changed.